### PR TITLE
Make DataNode release rather than delete when reassign

### PR DESCRIPTION
**Pull request by XuanYang-cn**
1. Reassgin now will assign to the original Node if no other nodes
   avaliable
2. Make AddNode balance async: ToRealse + Reassign

See also: #16114, #17270

Signed-off-by: yangxuan <xuan.yang@zilliz.com>

### CODE REVIEW

File: internal/datacoord/channel_manager.go [hunk 1]
```diff
@@ -348,12 +353,26 @@ func (c *ChannelManager) DeleteNode(nodeID int64) error {
...
 	}
-	_, err := c.store.Delete(nodeID)
-	return err
+
+	channels, err := c.store.Delete(nodeID)
+	if err != nil {
+		return err
+	}
+
+	chNames := make([]string, len(channels))
```
Comment (congqixia) [inline internal/datacoord/channel_manager.go:369]: maybe `chNames := make([]string, 0, len(channels))`

File: internal/datacoord/channel_manager.go [hunk 2]
```diff
@@ -318,17 +323,17 @@ func (c *ChannelManager) AddNode(nodeID int64) error {
...
+		log.Info("register node with no reassignment", zap.Int64("registered node", nodeID))
 		return nil
 	}
 
 	log.Info("register node",
 		zap.Int64("registered node", nodeID),
 		zap.Array("updates", updates))
 
-	return c.updateWithTimer(updates, datapb.ChannelWatchState_ToWatch)
+	return c.updateWithTimer(updates, datapb.ChannelWatchState_ToRelease)
```
[the remark this answers is not in the document]
Comment (XuanYang-cn) [reply internal/datacoord/channel_manager.go:336]: yes, The updates generated by registerPolicy now only consist of  those channel-node pairs need to be released

**Review by congqixia — COMMENTED** [review]
/lgtm

**Pull request merged**